Makefile

```diff
@@ -1,44 +1,45 @@
 OBJS := Parser.o Game.o Unit.o main.o
-CLFAGS := -Wall -Werror -std=c++17
+CFLAGS := -Wall -Werror -std=c++17
 RUN := g++
 
-CPPCHCK := cppcheck
-CPPCHCKOBJS := Parser.cpp Game.cpp Unit.cpp main.cpp
-CPPCHCKFLAGS := --enable=warning --error-exitcode=1
-CPPCHCKUPFLAGS := --enable=all --output-file=cppreport.txt
-VALGRINDOBJS := ./runGame test/units/unit1.json test/units/unit2.json 
-VALGRINDFLAGS := --error-exitcode=1 --leak-check=full
+SCA := cppcheck
+SCAOBJS := Parser.cpp Game.cpp Unit.cpp main.cpp
+SCAFLAGS := --enable=warning --error-exitcode=1
+SCAUPFLAGS := --enable=all --output-file=cppreport.txt
+MEMCHECK := valgrind
+MEMCHECKOBJS := ./runGame test/units/unit1.json test/units/unit2.json 
+MEMCHECKFLAGS := --error-exitcode=1 --leak-check=full
 DIFFOBJS := output.txt test/units/good_output.txt
 CMAKEOBJ := CMakeLists.txt
 SUBDIR := test
 
 .DEFAULT_GOAL: runGame
-.PHONY: cppcheck cppcheckup valgrind diff doc buildunittest unittest
-tests: runGame cppcheck cppcheckup valgrind diff
+.PHONY: sca scaup memcheck diff doc buildunittest unittest
+tests: runGame sca scaup memcheck diff
 
 runGame: $(OBJS)
 	$(RUN) $(CFLAGS) -o runGame $(OBJS)
 
-Parser.o: Parser.cpp
-	$(RUN) $(CFLAGS) -c Parser.cpp Parser.h
+Parser.o: Parser.cpp Parser.h
+	$(RUN) $(CFLAGS) -c Parser.cpp
+	
+Unit.o: Unit.cpp Unit.h Parser.h
+	$(RUN) $(CFLAGS) -c Unit.cpp
 
-Game.o: Game.cpp
-	$(RUN) $(CFLAGS) -c Game.cpp Game.h
+Game.o: Game.cpp Game.h Unit.h
```

Comment:
es a unit.h nem rantja be a parser.h-t? (be kellene neki)

Reply:
akkor mindenhova meg kell adni az "alatta" lévő függőséget? tehát mainhez az összes header, game-hez unit és parser, és így tovább?

Reply:
igy-igy, szepen "rekurzivan". Jo moka, ugye? :-D Na ezert nem szoktunk magaban make-et hasznalni, majd jon a `cmake` video :-)

Reply:
akkor mostmár tényleg megvan 😅